alacritty_terminal/src/config/window.rs

```diff
@@ -34,13 +36,35 @@ pub struct WindowConfig {
 
     /// Window class
     #[serde(deserialize_with = "failure_default")]
-    pub class: Option<String>,
+    pub class: Class,
+
+    /// GTK theme variant
+    #[serde(deserialize_with = "deserialize_gtk_theme_variant")]
+    pub gtk_theme_variant: Option<String>,
 
     /// TODO: DEPRECATED
     #[serde(deserialize_with = "failure_default")]
     pub start_maximized: Option<bool>,
 }
 
+fn deserialize_gtk_theme_variant<'de, D>(deserializer: D) -> Result<Option<String>, D::Error>
```

Comment:
You've copied the working directory deserializer here, right? I think it should be possible to make this a generic `deserialize_option` with the Result being `Result<Option<T>, D::Error>`. If `Deserialize` is a trait bound for `T`, then we can just use it for every `Option` which we want to accept explicit nones for.